lib/experimental/fonts/font-library/font-library.php

```diff
@@ -132,3 +132,30 @@ function wp_get_font_dir( $defaults = array() ) {
 		return apply_filters( 'font_dir', $defaults );
 	}
 }
+
+// @core-merge: This code should probably go into ...?
+if ( ! function_exists( 'wp_use_font_library' ) ) {
+	/**
+	 * Function that determines if the font library should be activated in the site editor.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @param array                   $editor_settings The editor settings to be potentially filtered.
+	 * @param WP_Block_Editor_Context $editor_context  The editor context.
+	 */
+	function wp_use_font_library( $editor_settings, $editor_context ) {
+		/**
+		 * Filters whether a post is able to be edited in the block editor.
+		 *
+		 * @since 6.5.0
+		 */
+		$use_font_library = apply_filters( 'use_font_library_for_site_editor', true );
+
+		if ( isset( $editor_context->name ) && 'core/edit-site' === $editor_context->name && ! $use_font_library ) {
+			$editor_settings['disableFontLibrary'] = true;
+		}
+
+		return $editor_settings;
+	}
+	add_filter( 'block_editor_settings_all', 'wp_use_font_library', 10, 2 );
+}
```

Comment:
If it's already possible to filter block_editor_settings_all to turn off the font library UI, I don't think we need this `wp_use_font_library` functional at all. Those wanting to disable it can use this filter directly.

We should include this in the dev note.

Reply:
Great point. I removed it entirely and updated the testing instructions for this PR to use the `block_editor_settings_all` filter directly. We can include that bit in the dev note.